Include/internal/pycore_object.h

```diff
@@ -29,6 +29,9 @@ _PyType_HasFeature(PyTypeObject *type, unsigned long feature) {
 
 extern void _PyType_InitCache(PyInterpreterState *interp);
 
+PyAPI_FUNC(PyObject *) _PyType_DisabledNew(PyTypeObject *type, PyObject *args,
+                                           PyObject *kwds);
```

Comment:
What do you think of adding it to the public C API but with "_Py" prefix to mark it as a private function? It's a little bit annoying to have to define the Py_BUILD_CORE_MODULE macro.

Maybe tomorrow, we can even add it to the public C API.

Reply:
Yes, that's what I suggested in https://github.com/python/cpython/pull/25653#discussion_r622186382

Reverting f74d5749b44d422880b12f02eb93279ca0d27c54 would achieve that.